### ISSUE TITLE
Unable to generate install-config.yml for Openstack

### ISSUE THREAD

**Issue by baba-tunde** [openshift/okd#418]
**Describe the bug**
Unable to generate install-config.yml for Openstack
`openshift-install create install-config --log-level=debug --dir=./` always ends up with the error message `FATAL failed to fetch Install Config: failed to generate asset "Install Config": failed to generate OpenStack cloud info: failed to fetch machine subnet info: Resource not found`

**Version**
4.6.0-0.okd-2020-11-27-200126

**How reproducible**
100%

**Log bundle**
```
openshift-install create install-config --log-level=debug --dir=./
DEBUG OpenShift Installer 4.6.0-0.okd-2020-11-27-200126 
DEBUG Built from commit 7b5bd8be9e5f18cc9b8af5f95d629ddd64c015cc 
DEBUG Fetching Install Config...                   
DEBUG Loading Install Config...                    
DEBUG   Loading SSH Key...                         
DEBUG   Loading Base Domain...                     
DEBUG     Loading Platform...                      
DEBUG   Loading Cluster Name...                    
DEBUG     Loading Base Domain...                   
DEBUG     Loading Platform...                      
DEBUG   Loading Pull Secret...                     
DEBUG   Loading Platform...                        
DEBUG   Fetching SSH Key...                        
DEBUG   Generating SSH Key...                      
? SSH Public Key /home/ubuntu/.ssh/rsa.pub
DEBUG   Fetching Base Domain...                    
DEBUG     Fetching Platform...                     
DEBUG     Generating Platform...                   
? Platform openstack
? Cloud shiftstack
? ExternalNetwork ext_net
? APIFloatingIPAddress *.*.*.*
? FlavorName c4.m16
DEBUG   Generating Base Domain...                  
? Base Domain example.org
DEBUG   Fetching Cluster Name...                   
DEBUG     Fetching Base Domain...                  
DEBUG     Reusing previously-fetched Base Domain   
DEBUG     Fetching Platform...                     
DEBUG     Reusing previously-fetched Platform      
DEBUG   Generating Cluster Name...                 
? Cluster Name okd-cluster
DEBUG   Fetching Pull Secret...                    
DEBUG   Generating Pull Secret...                  
? Pull Secret [? for help] ******************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************
DEBUG   Fetching Platform...                       
DEBUG   Reusing previously-fetched Platform        
DEBUG Generating Install Config...                 
FATAL failed to fetch Install Config: failed to generate asset "Install Config": failed to generate OpenStack cloud info: failed to fetch machine subnet info: Resource not found
```

[openshift_install.log](https://github.com/openshift/okd/files/5661418/openshift_install.log)

**Comment by vrutkovs** [comment]
@mandre hey, do you know why would installer reject a network config with this error message?

**Comment by mandre** [comment]
Hi @baba-tunde, it looks like you're setting the [`platform.openstack.machinesSubnet`](https://github.com/openshift/installer/blob/release-4.6/docs/user/openstack/customization.md#cluster-scoped-properties)  property in your `install-config.yaml` and this points to a resource that does not exist in your cloud.
This needs to be the UUID of an existing subnet. Check the [Custom Subnets](https://github.com/openshift/installer/blob/release-4.6/docs/user/openstack/customization.md#custom-subnets) documentation for more info on how to use the feature.

/assign

**Comment by baba-tunde** [comment]
Hi @mandre,
Is this something new ? On 4.5 it was working fine.
In fact I am following this doc https://docs.okd.io/latest/installing/installing_openstack/installing-openstack-user.html
With 4.6 should I need to define the Machines Subnet before starting the process ?

**Comment by mandre** [comment]
Could your perhaps share the `platform.openstack` section of your `install-config.yaml` file? Do you confirm that the value you pass to platform.openstack.machinesSubnet is the UUID of an existing subnet?

The validation was a bit re-worked on 4.6 and might get a slightly different error message, but I do not see how this would have worked previously if you tried passing an non-existing machine subnet.

**Comment by baba-tunde** [comment]
@mandre,
I don't have any install-config.yaml file. The first command that produce this file failed with the error I described above.
openshift-install create install-config --log-level=debug --dir=./

**Comment by baba-tunde** [comment]
on 4.5 I used to create all the network afterward with the ansible playbooks.

**Comment by mandre** [comment]
Oh I see you're running the `openshift-install create install-config` command, sorry I got confused.

It looks like your cloud returns a generic ErrDefault404 error rather than the [ErrResourceNotFound error we expected](https://github.com/openshift/installer/blob/6e5ef40/pkg/asset/installconfig/openstack/validation/cloudinfo.go#L135-L139).

Do you know what version of OpenStack you're deploying on?

**Comment by mandre** [comment]
https://github.com/openshift/installer/pull/4474 should fix it for 4.7, we'll then backport to 4.6.

**Comment by baba-tunde** [comment]
Version 15

**Comment by baba-tunde** [comment]
Hi @mandre Is there a way to work around this ?

**Comment by mandre** [comment]
> Hi @mandre Is there a way to work around this ?

Unfortunately, no. I'm afraid that your options are to to either wait for the fix to be included in the 4.6 branch of the installer, or apply [the patch fixing the issue](https://github.com/openshift/installer/pull/4474) to the release-4.6 branch yourself and [rebuild the installer](https://github.com/openshift/installer#quick-start).

One last option would be to install 4.5 and upgrade to 4.6.

**Comment by wjiangjay** [comment]
@baba-tunde hi, is https://github.com/openshift/installer/pull/4474 work well for you?
